### PR TITLE
Tic Tac Toe environment

### PR DESCRIPTION
**Pull request by rajivsambharya**
tictactoe environment added with tests

### CODE REVIEW

**Comment by thomas-a-neil** [comment]
Disregard the travis CI stuff, I was just messing around with it.

File: tictactoe_env.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        return np.array(legal_actions)
+
+    def is_game_over(self, state):
+        result = self.outcome(state)
+        if result == 2:
+            return 0
+        else:
+            return 1
+
+    def outcome(self, state):
```
Comment (thomas-a-neil) [inline tictactoe_env.py]: So only x can win in this environment? Shouldn't this return `-1` sometimes?

File: tictactoe_env.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
+import numpy as np
+
+
+class TicTacToeEnv(object):
+    def __init__(self):
+        """
+        state space: 2x3x3
+        action space: 18 possible values -- 1 hot
```
Comment (Rulison) [inline tictactoe_env.py]: I think the action space should be only 3x3.  In AlphaGo there is no color in the action space.

File: tictactoe_env.py (hunk 1 above)
Comment (rajivsambharya) [reply tictactoe_env.py]: omg you are totally right.  I'll fix.

File: tictactoe_env.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        num_os = state[1, :, :].sum()
+        if num_os == num_xs:
+            return 0
+        elif num_xs > num_os:
+            return 1
+        else:
+            return -1
+
+
+    def convert_turn_to_winner(self, turn):
```
Comment (thomas-a-neil) [inline tictactoe_env.py]: Call me silly, but how about lets make 

0 indicate x's turn, and if x won
1 indicate o's turn, and if o won.

-1 can be a draw
2 can be game is not over.

And then we can just return the `turn` directly and not do this somewhat confusing algebra

Comment (thomas-a-neil) [reply tictactoe_env.py]: Oh I see, this emulates the -1 and 1 of the chess environment (and what is expected by the `v` vector, I think). 

I still think this expression is overkill but whatever

File: tictactoe_env.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        return (1- turn) * 2 - 1
+
+
```
Comment (thomas-a-neil) [inline tictactoe_env.py]: Add this to your sublime settings:

```
"trim_trailing_white_space_on_save": true
```

**Pull request merged**